packages/salesforcedx-utils-vscode/src/cli/sfdxProjectJsonParser.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { SfdxProject } from '@salesforce/core';
+import { JsonArray, JsonMap } from '@salesforce/ts-types';
+
+import * as path from 'path';
+
+export class SfdxProjectJsonParser {
+  public async getValue(workspacePath: string, key: string) {
+    const sfdxProject = await SfdxProject.resolve(workspacePath);
+    const sfdxProjectJson = await sfdxProject.resolveProjectConfig();
+    return sfdxProjectJson[key];
```

Comment:
This is awesome. I tried to break it by deleting `sfdcLoginUrl` on my `sfdx-project.json` file and it turns out that the cli has a default for this type of scenarios.

Reply:
Yeah I was hoping that would be the case and it looks like it is! I'm taking advantage of it when determining to render the "Project Default" option with the org type selection.